app/controllers/consent_responses_controller.rb

```diff
@@ -11,21 +11,33 @@ def new
   end
 
   def create
+    health_questions = ConsentResponse::HEALTH_QUESTIONS
+            .fetch(:hpv)
+            .map { |question| { question: } }
+
     if consent_response_who_params.present?
       @draft_consent_response.assign_attributes(
         consent_response_who_params.merge(
           campaign: @session.campaign,
           route: "phone",
-          health_questions: ConsentResponse::HEALTH_QUESTIONS
-            .fetch(:hpv)
-            .map { |question| { question: } }
+          health_questions:
         )
       )
       if @draft_consent_response.save(context: :edit_who)
         redirect_to action: :edit_consent
       else
         render :edit_who
       end
+    else
+      # If the params are missing, assume this is the Gillick competence route.
+      # This feels like it could be more explicit.
+      @draft_consent_response.update!(
+        campaign: @session.campaign,
```

Comment:
Should this be set in `set_draft_consent_response`? Would there be any situation we don't want it set?

Reply:
Come to think of it, when finding the draft record in `set_draft_consent_response` it doesn't specify the campaign ... it probably should.